File: kata-solutions/stack-kata/stack-kata-java/app/src/main/java/Stack.java
public class Stack {
  private Plate[] plates = new Plate[2];
  int size = 0;
  
  public boolean isEmpty() {
    return this.size == 0;
  }

  public void push(Plate plate) {
     
     this.plates[this.size++] = plate;
  }

  public Plate pop() {
     if (isEmpty()) {
       throw new RuntimeException("Stack underflow");
     }
    
     return this.plates[--this.size];
  }
}

File: kata-solutions/stack-kata/stack-kata-typescript/src/Stack.ts
'use strict';

export class Stack {
  size: number = 0
  element: Array<number> = new Array()

  isEmpty(): boolean {
    return this.size == 0
  }

  pop(): number {
    if (this.isEmpty())
      throw new Error("Stack underflow")
    
    return this.element[--this.size]
  }

  push(newElement: number): void {
    this.element[this.size++] = newElement
  }
}
